src/devices/Tlc1543/Tlc1543.cs

```diff
@@ -0,0 +1,239 @@
+            }
+
+            int value = 0;
+            _digital.Write(_chipSelect, 0);
+            for (int i = 0; i < 10; i++)
+            {
+                if (i < 4)
+                {
+                    // send 4-bit Address
+                    if ((channelNumber >> (3 - i) & 0x01) != 0)
+                    {
+                        _digital.Write(_address, 1);
+                    }
+                    else
+                    {
+                        _digital.Write(_address, 0);
+                    }
+                }
+
+                // read 10-bit data
+                value <<= 1;
+                if (_digital.Read(_dataOut) == PinValue.High)
+                {
+                    value |= 0x01;
+                }
+
+                _digital.Write(_inputOutputClock, 1);
+                _digital.Write(_inputOutputClock, 0);
+            }
+
+            _digital.Write(_chipSelect, 1);
+            // Max conversion time (21us) as seen in table on page 10 in TLC1543 documentation
+            DelayHelper.Delay(_conversionTime, false);
+            return value;
+        }
+
+        /// <summary>
+        /// Reads sensor value in Fast Mode 1 (2 cycle used)
+        /// <br>First cycle: Ask for value on the channel (10 clock transfer with !chipSelect high)</br>
+        /// <br>Second cycle: Return value from the channel while charging to <seealso cref="ChargeChannel"/></br>
+        /// </summary>
+        /// <param name="channelNumber">Channel to be read</param>
+        /// <returns>A 10 bit value corresponding to relative voltage level on specified device channel</returns>
+        public int ReadChannel(Channel channelNumber)
+        {
+            ReadChannel((int)channelNumber);
+            return ReadChannel((int)ChargeChannel);
+        }
+
+        /// <summary>
+        /// Reads sensor values into an List
+        /// <remarks>
+        /// <br>Fastest and most efficient to use</br>
+        /// <br>Reads channels in order in the list</br>
+        /// <br>Uses (list size + 1) cycles </br>
+        /// </remarks>
+        /// </summary>
+        /// <param name="channelList">List of channels to read</param>
+        /// <returns>List of 10 bit values corresponding to relative voltage level on specified device channels</returns>
+        public List<int> ReadChannels(List<Channel> channelList)
+        {
+            List<int> values = new List<int>(channelList.Count);
+            ReadChannel((int)channelList[0]);
+            for (int i = 1; i < channelList.Count; i++)
+            {
+                values.Add(ReadChannel((int)channelList[i]));
+            }
+
+            values.Add(ReadChannel((int)ChargeChannel));
+            return values;
+        }
+
+        /// <summary>
+        /// Reads all sensor values into an List
+        /// </summary>
+        /// <returns>List of 10 bit values corresponding to relative voltage level on all channels</returns>
+        public List<int> ReadAll()

```

Comment:
considering this is not hardware supported I'll vote to remove this helper. As I mentioned below we should not allocate in APIs like this (also this will be hard to support on nano since it doesn't support `List<T>`)